Jint/Native/Array/ArrayInstance.cs

```diff
@@ -120,14 +113,14 @@ public override bool DefineOwnProperty(string propertyName, PropertyDescriptor d
 
                 if (_array.Count < oldLen - newLen)
                 {
-                    var keys = _array.Keys.ToArray();
+                    var keys = new List<uint>(_array.Keys);
```

Comment:
You need the `keys` instance because you might be mutating the keys, but it never escape this method.
This can be cached using a thread static instance that is shared globally. On each call, we clear the instance, then fill it, then actually use the value. It will mean that you'll have much fewer allocations

Reply:
Yes I was wondering about thread safety and reuse here. I guess multiple simultaneous parser instances can be common (cannot use plain static). Need to adjust for thread-static one then. Same goes for FastDtoa.NumberToString .